src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -57,13 +57,11 @@ public Task StartAsync(CancellationToken cancellationToken)
 
             if (_poll == null)
             {
-                var errorText = string.Format("Path '{0}' is invalid. IFolderItem.RootPath must be set to a valid directory location.", _folderSource.Path);
-                _trace.Error(errorText);
-
-                throw new InvalidOperationException(errorText);
+                var errorText = string.Format("Path '{0}' is invalid. Path must be set to a valid directory location.", _folderSource.Path);
+                var ex = new InvalidOperationException(errorText);
+                _trace.Error(errorText, ex);
```

Comment:
GetFolderreferenceAsync does not hit the backend and just returns an IFolder item same design as storage SDK. ListAsync returns list of the files which not sure why it can be used here.

The only way to check if the path is valid before creating the FileWatcher(i.e. saving the json file) is to have the portal call FolderExists when a path is entered in the 'Path' textbox. but then it means the portal will need to take a dependency on ApiHub SDK and all the complexities which come with it.